packages/react-instantsearch-hooks-dom/src/ui/__tests__/ToggleRefinement.test.tsx

```diff
@@ -0,0 +1,128 @@
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { ToggleRefinement } from '../ToggleRefinement';
+
+import type { ToggleRefinementProps } from '../ToggleRefinement';
+
+describe('ToggleRefinement', () => {
+  function createProps(
+    props: Partial<ToggleRefinementProps>
+  ): ToggleRefinementProps {
+    return {
+      label: 'Free shipping',
+      onChange: jest.fn(),
+      classNames: {},
+      ...props,
+    };
+  }
+
+  test('renders with props', () => {
+    const props = createProps({});
+    const { container } = render(<ToggleRefinement {...props} />);
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="ais-ToggleRefinement"
+        >
+          <label
+            class="ais-ToggleRefinement-label"
+          >
+            <input
+              class="ais-ToggleRefinement-checkbox"
+              type="checkbox"
+            />
+            <span
+              class="ais-ToggleRefinement-labelText"
+            >
+              Free shipping
+            </span>
+          </label>
+        </div>
+      </div>
+    `);
+  });
+
+  test('customizes the label', () => {
+    const props = createProps({ label: 'Gluten free' });
+    const { container } = render(<ToggleRefinement {...props} />);
+
+    expect(
+      container.querySelector('.ais-ToggleRefinement-labelText')
+    ).toHaveTextContent('Gluten free');
+  });
+
+  test('sets the checkbox as checked', () => {
+    const props = createProps({ checked: true });
+    const { container } = render(<ToggleRefinement {...props} />);
+
+    expect(
+      (
+        container.querySelector(
+          '.ais-ToggleRefinement-checkbox'
+        ) as HTMLInputElement
+      ).checked
+    ).toBe(true);
+  });
+
+  test('passes an `onChange` callback to the checkbox', () => {
+    const props = createProps({});
+    const { container } = render(<ToggleRefinement {...props} />);
+
+    userEvent.click(
+      container.querySelector<HTMLInputElement>(
+        '.ais-ToggleRefinement-checkbox'
+      )!
+    );
+
+    expect(props.onChange).toHaveBeenCalledTimes(1);
+    expect(props.onChange).toHaveBeenLastCalledWith(true);
+  });
+
+  test('customizes the class names', () => {
+    const props = createProps({
+      classNames: {
+        root: 'MyCustomToggleRefinement',
+        label: 'MyCustomToggleRefinementLabel',
+        checkbox: 'MyCustomToggleRefinementCheckbox',
+        labelText: 'MyCustomToggleRefinementLabelText',
```

Comment:
in the others I have this as the uppercase version of the label (so it's easily to visually see in the snapshot). I'm not sure if we need to be consistent here though, it's just a test

Reply:
I like consistency too, and I agree that capitals are better because they stick out.

Updated in db1b19a0.